mantidproject/mantid
# Ensure all tests run under Qt5

**Description of work.**

During the removal of Qt4 targets some tests were accidentally removed from running. This reinstates the ones that are not for targets that will be removed. It also removes the general test target for ScientificInterfaces in favour of each library having its own test executable. To this end the reminaing Muon & all ISIS Reflectometry tests have moved to subfolders of their respective libraries.

I also fixed a warning in `qt/scientific_interfaces/ISISReflectometry/test/Options/OptionsDialogPresenterTest.h` that clang gave me about a defaulted constructor being deleted.



**To test:**

* Code review
* Check that all tests run and contain the ISISRelfectometry & Muon tests from master.

Fixes #30423 

*This does not require release notes* because **it is a set of internal changes**



---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
